pac4j-core/src/main/java/org/pac4j/core/profile/InternalAttributeHandler.java

```diff
@@ -0,0 +1,128 @@
+package org.pac4j.core.profile;
+
+import org.pac4j.core.profile.converter.Converters;
+import org.pac4j.core.util.CommonHelper;
+import org.pac4j.core.util.JavaSerializationHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * Internally handles attributes (set / get).
+ *
+ * @author Jerome Leleu
+ * @since 2.0.0
+ */
+public class InternalAttributeHandler implements Serializable {
```

Comment:
This component was added to allow some transformation on the attributes before saving them or retrieving them. Currently, it's only for CAS, but this could be enhanced in some other ways. Tough, as it seems unlikely, it's mostly an internal component. Hence the name.